packages/multichain/src/adapters/caip-permission-adapter-session-scopes.ts

```diff
@@ -56,19 +68,23 @@ const getNormalizedScopesObject = (
       let notifications: string[] = [];
 
       if (namespace === KnownCaipNamespace.Wallet) {
-        if (reference) {
-          methods =
-            KnownWalletNamespaceRpcMethods[
-              reference as NonWalletKnownCaipNamespace
-            ] ?? [];
-        } else {
+        if (!reference) {
           methods = KnownWalletRpcMethods;
+        } else if (reference === KnownCaipNamespace.Eip155) {
+          methods = KnownWalletNamespaceRpcMethods[reference];
+        } else {
+          methods = isCaipChainId(scopeString)
+            ? getNonEvmSupportedMethods(scopeString)
+            : [];
```

Comment:
@FrederikBolding just wondering if we've considered what `wallet:<caip-2 reference>` scoped snaps would mean/look like. Not saying this case won't exist, just curious if we've thought about it.